DEVELOPER_GUIDE.md

```diff
@@ -3,74 +3,185 @@
+  - [Next Steps](#next-steps)
+- [Alternative development installations](#alternative-development-installations)
+  - [Optional - Run OpenSearch with plugins](#optional---run-opensearch-with-plugins)
+  - [Alternative - Run OpenSearch from tarball](#alternative---run-opensearch-from-tarball)
+  - [Configure OpenSearch Dashboards for security](#configure-opensearch-dashboards-for-security)
+- [Building artifacts](#building-artifacts)
+  - [Building the Docker image](#building-the-docker-image)
+- [Code guidelines](#code-guidelines)
+  - [General](#general)
+  - [HTML](#html)
+  - [SASS files](#sass-files)
+  - [TypeScript/JavaScript](#typescriptjavascript)
+  - [React](#react)
+  - [API endpoints](#api-endpoints)
 
-Besides the content in this developer guide, the following developer guides may also apply to all development within the OpenSearch Dashboards project. Please make sure to also read them:
+## Getting started guide
 
-- [Accessibility developer guide (EUI Docs)](https://elastic.github.io/eui/#/guidelines/accessibility)
-- [SASS developer guide (EUI Docs)](https://elastic.github.io/eui/#/guidelines/sass)
+This guide is for any developer who wants a running local development environment where you can make, see, and test changes. It's opinionated to get you running as quickly and easily as possible, but it's not the only way to set up a development environment.
 
-## Getting Started
+If you're only interested in installing and running this project, please see the [Installing OpenSearch Dashboards](https://opensearch.org/docs/latest/install-and-configure/install-dashboards) instead.
 
-If you would like to install and run this project, please see the [Downloads Page](https://opensearch.org/downloads.html).
+If you're planning to contribute code (features or fixes) to this repository, great! Make sure to also read the [contributing guide](CONTRIBUTING.md).
 
-#### Prerequisites
+### Key technologies
 
-We recommend using [Node Version Manager](https://github.com/nvm-sh/nvm) to install
-the node version we need.
+OpenSearch Dashboards is primarily a node.js web application built in React. To effectively contribute you should be familiar with HTML ([usage guide](#html)), SASS styling ([usage guide](#sass-files)), TypeScript and JavaScript ([usage guide](#typescriptjavascript)), and React ([usage guide](#react)).
 
-### Bootstrap OpenSearch Dashboards
+### Prerequisites
+
+To develop on OpenSearch Dashboards, you'll need:
+
+- A [GitHub account](https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account)
+- [`git`](https://git-scm.com/) for version control
+- [`node`](https://nodejs.org/), [`npm`](https://www.npmjs.com/), and [`yarn`](https://yarnpkg.com/) for building and running the project
+- A code editor of your choice, configured for JavaScript/TypeScript. If you don't have a favorite editor, we suggest [Visual Studio Code](https://code.visualstudio.com/)
```

Comment:
Re-read - no IDE required. However, contributors _do_ need some way to edit code, even if it's the GH UI.

For beginners in particular (the target audience), the goal is to avoid decision paralysis. We absolutely should provide a suggestion for folks who otherwise have no preferences, and it makes sense to suggest one most maintainers are familiar with.

Is it the "configured for JavaScript/TypeScript" you take issue with? Again, as a maintainer, I want to make sure contributors have the tooling in place that makes it the most likely that we get clean contributions, and integrated linting/language support can help folks who don't already have a preferred workflow and toolchain.

Reply:
I didn't see "no IDE Required". However, re-reading, I find the choice of words, ie "code editor", pretty un-opinionated.

Re the suggestion, I would feel less agitated if the text was "If you don't have a favorite editor,  [Visual Studio Code](https://code.visualstudio.com/) is a free and very capable one."

Reply:
Happy to make that change in a follow-up.